### ISSUE TITLE
StructureViewer crashes when Selection is out of bounds

### ISSUE THREAD

**Issue by danielhollas** [aiidalab/aiidalab-widgets-base#413]
## Automated report

_This issue was created with the app's automated bug reporting feature.
Attached to this issue is the full traceback as well as an environment
fingerprint that contains information about the operating system as well as all
installed libraries._

## Additional comments (optional):

To reproduce, go to the Selection tab, and try to select for an atom that is not there, e.g. atom number 10 in the methane molecule. The exception appears when you click the "Apply selection" button

![obrazek](https://user-images.githubusercontent.com/9539441/206879083-ef8afc50-03ad-4ab6-acdf-9f8b42f0dcd4.png)

## Attachments

<details>
<summary>Traceback</summary>

```python-traceback
/opt/conda/lib/python3.8/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    684                 # Send new state to front-end
    685                 self.send_state(key=name)
--> 686         super(Widget, self).notify_change(change)
    687 
    688     def __repr__(self):

/opt/conda/lib/python3.8/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1384     def notify_change(self, change):
   1385         """Notify observers of a change event"""
-> 1386         return self._notify_observers(change)
   1387 
   1388     def _notify_observers(self, event):

/opt/conda/lib/python3.8/site-packages/traitlets/traitlets.py in _notify_observers(self, event)
   1429                 c = getattr(self, c.name)
   1430 
-> 1431             c(event)
   1432 
   1433     def _add_notifiers(self, handler, name, type):

~/apps/aiidalab-widgets-base/aiidalab_widgets_base/viewers.py in _observe_selection_2(self, _)
   1061     @observe("selection")
   1062     def _observe_selection_2(self, _=None):
-> 1063         self.selection_info.value = self.create_selection_info()
   1064 
   1065 

~/apps/aiidalab-widgets-base/aiidalab_widgets_base/viewers.py in create_selection_info(self)
   1002         # Find geometric center.
   1003         geom_center = print_pos(
-> 1004             np.average(self.structure[self.selection].get_positions(), axis=0)
   1005         )
   1006 

/opt/conda/lib/python3.8/site-packages/ase/atoms.py in __getitem__(self, i)
   1073         atoms.arrays = {}
   1074         for name, a in self.arrays.items():
-> 1075             atoms.arrays[name] = a[i].copy()
   1076 
   1077         atoms.constraints = conadd

IndexError: index 9 is out of bounds for axis 0 with size 5
```
</details>

<details>
<summary>Environment fingerprint</summary>
<pre>eNplWFtv67gR_iuB-7oyLMmxk7wt0u5DcdI9bfa0KIrigKJomTFFMiQVR1mc_97hzNBWuvCTPg7JuXxzoX9fvakQtbOrh_qnlTci
HVwYVw-_r0SQR52UTFNQq4f_rHbbTqfVT6u_fPll9V8QndPR2e-X3at2fbeuWxC4Qn-qt_vqWzfZNN08P329-SXom19luqm3N_X-
YbN_aJqbb7893jSbpln9gDOFPIlBRbxe615U0uXLV_V6t76HoxH87oIUAG4A3BTQiA6gplnv1_UCq0QaXfRKVmcXTvIotMWdn2SU
YXDdLGEd_fBH4bPuB5Vi1YlImm15l6u8PmW9duAIhsL4ip5p6QzTv6hxxjPvGVFjpyWec0eIjbo5ptEwhgZ63-sQ-bZtRnJs3lTl
RTricQ1tDoOzTSUPB52dUcO9m89w1WnbaztEWm94Pbgznt6scwDJMtD5nj9tLcPsk0OZW7oqpuB0n48BLXeIzFZWg7IqiOQCytZ4
fHKjlucAZCo24AkpoU0QsxrV6CD6UhjDBiGE96IyDfoU1BdhdsG6hGLkjs4oIbMjbtcUrc6d1BF126IJ3as3vKMGG7O6XXDJaE_R
2ONtj3CIenQWLDNFts7HybyQnDNZX4g3Xir_YIBUIWlyPRAaQtygHEUDnHFLUofhbUEK4GToVWLlN4xElSoL9BFGf6iAlvAdRzVq
e3CfFJc6yCmSysR_abQ8AbBnUuM3ZNPojUqUnBuI5HLJHvRQHbRRi9Sitei1tahFzhGMqXTGBTEKDsItQrYX6J-agodAxTldHYXt
DTBvwXW603UvbPwOMXfSSk4pKaLQHgNI_BuC8MdseAt2UxBmaYpm5KBHrEvEIeBvtrAXErjHZSTL9KqbBvRgRlpEoNIwbW-zkYgd
pqj699Gwqwn0QUHoyYclKr2OQKYO5Vq0488vwg4OReB3l2Vs9EW3hhO2n8xZS05hsCjvVKPQpnqDyPeXPCJpBcycvdM2UayzA1RS
LSUrBlO9S-WzckNwk8e9QKogc_E8QM4Krym8t2h2hl6iszHTSpBBO7wrE8E45FDLZegAmXFNgpb8cZioP-
QAUNnDIL3pj2JUjZhS1iDLwbd4te7BHH2Y8dJb9JnuraDMyB-jdyE7tRpVEuCKT9y6rgYV3RQklpZbyDlc9XPewKWa1EXdYOGkgl
UGy3RNTtCXsADXtuWADOVqNiVt4qIk5SWwhhE-9HrTHnjyGazUe1K2Vz0bn_2hI2hPVCMHvaheFyfm779q-yIaCgj56wWRKulRLb
R5cR3xrmYm5mgiRbhqtIwFdaBt_H2JeFtOmsAwFSpIeQgM-Qe8vVyhXrzNGtVXfNEUFxae9PlaXG8ZiM68cVpT4TYinKBCQLkLpe
BuEE5AZAdJiTaQi4z4mCsoFjCRVD6493nRITNTL6WrprA8iZNbJM8TXNW7M00q2ws0-WdxUItrRpFyryCvtmuiyBWstIUqxhfRmp
SiK0UTT4BiMLHIHZo5QubMpS1A_9ghFlQFZSmUlILAE_lGP1JbL6qPcchVlLMZD5xM0r2WFCeK6jjZU1AlBlnKdlBhweHENFLOdn
m-EwRRQ7Iqpgq7t3YcmxbhlGemd27xOYZ2MG9andExGxKCICn7xpUaT3NJdVDEUTPqDHYaudg2lNvOqEu0sOY5qFmj6C91ZLPAro
zkrlrnYkZthfpJQ4HwME2lJDqjuL1uEBxfPWUCf0NTIie1qFwGnAR1IBSRBxwShP6mT67kIGPRcVzpxnQs4Svjo-9CSWzcooJ2OV
KsV81x8OodptKE-UT2-uFus9mQm-7pqCFOizrjBxigEvMCd0BzNirCuKAWifZVG4Pj3P1FzNOUQtwpMz6PlA2PAcjveVFEvZmGgT
K4plEFkJFzumHr3Ox4lMBPoDTMGKNO1ONYKDgo4Uc1xWsoN5nsNS_6VGUvnnQqzMJtMZdeVghF4yydH5qKhsDF6OnTDOdAD4iLoY
gpV0I3d1BTSPctI1LowCWChL7Oj_rwj39hUCgJQMhFkTi7GbkULDARFYNuA71M8siL7vw6D6OiRt2UsuZnKB2pEIp2QipCkynkpn
z_Oj_Nz3__wu-EFpG_iUezZOf8q1f2-flLiSyBWTPKiexGzJMZkJguXr_je-MICRi5pNClz1BEI2cyCTHftpmzjOSuCL1VcWwafu
vwSgTOUE_flXSABWiXL5GncT7GvUgxek4b0j0FMfMAxRs_0C11ccG_f376gnQop3zg66opHSzPGCnMnNstIjgliFLKd3cIvk5gFy
nUkD8KVuGkz47fLRdGxwWY8zocZHt_t-fL8NxJjMqsZ8HzIkziyI4FvpZlSGzodFyq8lJVxudNHkvu8hVRakiJCt42wRbiNoQzty
lRowzky_IWjUqdfGkh9-s9QM8wgTS_BRGPi64Jj4zJl_q1K26MR2XyoH4UI59JKHhymnTPlMncijo_J_I0QRPoHo2K-p0cRbM2PK
IvE8oOCxRQ-2eTx48SKnwmXdHq22XkamlHfDWYc_zaQBuTCLncq1JMCAxQaascm7a9p1qQ4bn0H562oBhPecbgUo4vGkiR_EyYsR
OzmArw1hK9K2EhuZguz-4dQ--pmqzOj4ieinxmXya-6D34VybDMzqKu8IQKhUw5PMlO_p-7UfKOnI9JIZOhsarW25jiEHVU3Hx_I
dviNt3nDfzPzBl8sUNHzAjCUrZLDvZ_FoVUI80npGV2yOusxF5fM7jZx5JcP_kD1CkKw52To8cmCmY6zN1MTsADkFvqbZy8fingG
ScF53yTQOjoGfj-JAfChTss0jy2LuhuCiT4yxhwrz81YFSqsM3KM87DUHKgvL898aG_6k4H5Uif_OThYbV6v87c8uW8ixru4sfcZ
Ha9hkeN6n0lXzpe-qq67OOeDMDNRfP3A_tffl_bPXjx_8Ad1VwzQ==</pre>
</details>

**By submitting this issue I confirm that I am aware that this information can
potentially be used to determine what kind of calculation was performed at the
time of error.**

**Comment by yakutovicha** [comment]
I think this might be fixed as a part of https://github.com/aiidalab/aiidalab-widgets-base/pull/371. @superstar54 correct me if I am wrong.

**Comment by superstar54** [comment]
Yes, this is fixed in #371.
